mlflow/utils/search_utils.py

```diff
@@ -37,19 +36,41 @@ def filter_string(self):
     def search_expressions(self):
         return self._search_expressions
 
+    @classmethod
+    def _trim_ends(cls, string_value):
+        return string_value[1:-1]
+
+    @classmethod
+    def _is_quoted(cls, value, pattern):
+        if (value.startswith(pattern) and not value.endswith(pattern)) or \
+                (not value.startswith(pattern) and value.endswith(pattern)):
+            raise MlflowException("Mismatched quote types in argument {}" % value,
+                                  error_code=INVALID_PARAMETER_VALUE)
+        return value.startswith(pattern) and value.endswith(pattern)
+
     @classmethod
     def _trim_backticks(cls, entity_type):
-        if entity_type.startswith("`"):
-            assert entity_type.endswith("`")
-            return entity_type[1:-1]
+        """Remove backticks from identifier like `param`, if they exist."""
+        if cls._is_quoted(entity_type, "`"):
+            return cls._trim_ends(entity_type)
         return entity_type
 
+    @classmethod
+    def _strip_quotes(cls, value):
+        """Argument value is of type string and expected to have quotes."""
+        if cls._is_quoted(value, "'") or cls._is_quoted(value, '"'):
+            return cls._trim_ends(value)
+        raise MlflowException("Parameter value is either not quoted or unidentified quote "
```

Comment:
Can we say what the valid quote types are in this error message?

Reply:
👍